src/js/analyses/components/Create/ReferenceSelector.js

```diff
@@ -1,58 +1,119 @@
-import { map } from "lodash-es";
 import PropTypes from "prop-types";
 import React from "react";
 import styled from "styled-components";
-import { Label, NoneFoundBox } from "../../../base";
+import { Label, Input, Icon } from "../../../base";
 import { MultiSelector, MultiSelectorItem } from "../../../base/MultiSelector";
-import { Link } from "react-router-dom";
+import { useFuse } from "../../../base/hooks";
+import {
+    SampleInputContainer,
+    IconContainer,
+    ItemComponentsContainer,
+    CreateNoneFoundBox
+} from "../../../subtraction/components/Selector";
+import { SelectedAnalysesObject } from "./Selected";
+import { GridContainer, CreateHeader, AllSelectedBox } from "./WorkflowSelector";
+import { NameOverflowItem } from "./SubtractionSelector";
+import { intersectionWith, difference } from "lodash-es";
 
 const StyledReferenceSelectorItem = styled(MultiSelectorItem)`
-    > span:last-child {
+    span:last-child {
         margin-left: auto;
     }
+    background-color: white;
+    outline: 1px solid;
+    outline-color: ${props => props.theme.color.greyLight};
 `;
 
+const compare = (option1, option2) => {
+    return option1.reference.id === option2;
+};
+
+const formattedLine = item => {
+    return (
+        <>
+            <NameOverflowItem type="references">{item.reference.name}</NameOverflowItem>
+            <span>
+                Index Version <Label>{item.version}</Label>
+            </span>
+        </>
+    );
+};
+
 export const ReferenceSelectorItem = ({ id, name, version }) => (
-    <StyledReferenceSelectorItem value={id}>
-        <strong>{name}</strong>
+    <StyledReferenceSelectorItem value={id} requireCheckbox={false}>
+        <NameOverflowItem type="references">{name}</NameOverflowItem>
         <span>
             Index Version <Label>{version}</Label>
         </span>
     </StyledReferenceSelectorItem>
 );
 
 export const ReferenceSelector = ({ hasError, indexes, selected, onChange }) => {
-    const referenceComponents = map(indexes, index => (
-        <ReferenceSelectorItem
-            key={index.reference.id}
-            id={index.reference.id}
-            name={index.reference.name}
-            version={index.version}
-        />
-    ));
+    const clearSelected = object => {
```

Comment:
```suggestion
    const clearSelected = selectedReference => { //or selectedIndex, whichever is really getting passed here
```
Since you are defining a custom clearSelected for each resource type you can afford to be less generic in your naming